### PR TITLE
referece/tools: add document for deploying dm with binary

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? 

This PR adds the document for deploying DM with binary.

### What is the related PR or file link(s)? 

https://github.com/pingcap/docs-cn/pull/1880
https://github.com/pingcap/docs-cn/pull/1949

### Which version does your change affect? 

`dev`, `v2.1`, `v3.0`

### CODE REVIEW

File: dev/how-to/deploy/data-migration-with-binary.md [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+./bin/dmctl --encrypt "123456"
+```
+
+Then, you get the encrypted password as shown below. Record this encrypted value, which is used for deploying DM-worker in the following steps.
+
+```
+fCxfQ9XKCezSzuCD0Wf5dUD+LsKegSg=
+```
+
+You can deploy DM-worker by using command-line parameters or by using the configuration file.
```
Comment (WangXiangUSTC) [inline dev/how-to/deploy/data-migration-with-binary.md]: support two ways to set config, but  can't delpoy DM-worker only by command-line, some config can only set in config file

File: dev/how-to/deploy/data-migration-with-binary.md [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+
+```bash
+bin/dm-master -config conf/dm-master.toml
+```
+
+Now, a DM cluster is successfully deployed.
+
+### Create a data replication task
+
+Suppose that MySQL1 and MySQL2 have several sharded tales on each instance. These tables have the same structure and the same prefix "t" in their table names. The databases that they are in are named with the same prefix "sharding". In each of these tables, there is no conflict between the primary key and the unique key.
```
Comment (WangXiangUSTC) [inline dev/how-to/deploy/data-migration-with-binary.md]:
```In each of these tables, there is no conflict between the primary key and the unique key.
```
don't understand this

Comment (TomShawn) [reply dev/how-to/deploy/data-migration-with-binary.md]: 每张表的主键和唯一键没有冲突。

Comment (WangXiangUSTC) [reply dev/how-to/deploy/data-migration-with-binary.md]: 这些分表原来都是独立的表，现在需要同步到 tidb 的一个表中，所以应该是这些分表之间没有同样的主键值/唯一键的值

**Comment by WangXiangUSTC** [comment]
LGTM

File: dev/how-to/deploy/data-migration-with-binary.md [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+| MySQL1     | 192.168.0.1 |
+| MySQL2     | 192.168.0.2 |
+| TiDB       | 192.168.0.3 |
+| DM-master  | 192.168.0.4 |
+| DM-worker1 | 192.168.0.5 |
+| DM-worker2 | 192.168.0.6 |
+
+Enable the binlog on MySQL1 and on MySQL2. DM-worker1 replicates the data from MySQL1 and DM-worker2 replicates the data from MySQL2.
+
+Based on this scenario, the following sections of this document describe how to deploy the DM cluster.
```
Comment (yikeke) [inline dev/how-to/deploy/data-migration-with-binary.md]:
```suggestion
Based on this scenario, the following sections describe how to deploy the DM cluster.
```

File: dev/how-to/deploy/data-migration-with-binary.md [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+./bin/dm-worker --help
+```
+
+```
+Usage of worker:
+  -L string
+        Log level. Available values: "debug", "info" (default value), "warn", "error" or "fatal"
+  -V    The output version number
+  -checker-backoff-max duration
+        The longest waiting time for the automatic recovery after errors are checked in the task check module. The default value is "5m0s" which generally needs no change. It is recommended that you do not change this default value unless you have an in-depth understanding of this parameter.
```
Comment (yikeke) [inline dev/how-to/deploy/data-migration-with-binary.md]:
```suggestion
        The longest waiting time for the automatic recovery after errors are found in the task check module. The default value is "5m0s" which generally needs no change. It is not recommended to change this default value unless you have an in-depth understanding of this parameter.
```

File: dev/how-to/deploy/data-migration-with-binary.md [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+  -checker-backoff-rollback duration
+        The time interval for adjusting the waiting time of the automatic recovery in the task check module. The default value is "5m0s" which generally needs no change. It is recommended that you do not change this default value unless you have an in-depth understanding of this parameter.
```
Comment (yikeke) [inline dev/how-to/deploy/data-migration-with-binary.md]:
```suggestion
        The time interval for adjusting the waiting time of the automatic recovery in the task check module. The default value is "5m0s" which generally needs no change. It is not recommended to change this default value unless you have an in-depth understanding of this parameter.
```

File: dev/how-to/deploy/data-migration-with-binary.md [hunk 6]
```diff
@@ -0,0 +1,367 @@
...
+  -checker-check-enable
+        Enables or disables the task status check. When it is enabled, DM tries to resume the data replication tasks interrupted by errors. Default value: "true".
```
Comment (yikeke) [inline dev/how-to/deploy/data-migration-with-binary.md]:
```suggestion
        Enables or disables the task status check. When it is enabled, DM automatically tries to resume the data replication tasks interrupted by errors. Default value: "true".
```

File: dev/how-to/deploy/data-migration-with-binary.md [hunk 7]
```diff
@@ -0,0 +1,367 @@
...
+  -config string
+        The path of the configuration file
+  -log-file string
+        The path of log files
+  -print-sample-config
+        Prints the sample configuration
+  -purge-expires int
+        The expiration time of relay logs. DM-worker tries to delete the relay logs whose last modified time is out of this value. Unit: hour.
```
Comment (yikeke) [inline dev/how-to/deploy/data-migration-with-binary.md]:
```suggestion
        The expiration time of relay logs. DM-worker tries to delete the relay logs whose last modified time exceeds this value. Unit: hour.
```

File: dev/how-to/deploy/data-migration-with-binary.md [hunk 8]
```diff
@@ -0,0 +1,367 @@
...
+# Available values: "mysql", "mariadb"
+flavor = "mysql"
+
+# The path in which relay logs are stored
+relay-dir = "./relay_log"
+
+# The path in which the metadata of DM-worker is stored
+meta-dir = "dm_worker_meta"
+
+# Enables or disables gtid in the relay log unit
```
Comment (yikeke) [inline dev/how-to/deploy/data-migration-with-binary.md]:
```suggestion
# Enables or disables gtid in the relay log processing unit
```

File: dev/how-to/deploy/data-migration-with-binary.md [hunk 9]
```diff
@@ -0,0 +1,367 @@
...
+# remain-space = 15
+
+# Configurations of task status check
+# [checker]
+# check-enable = true
+# backoff-rollback = 5m
+# backoff-max = 5m
+```
+
+Then, run the following command in the terminal to operate DM-worker:
```
Comment (yikeke) [inline dev/how-to/deploy/data-migration-with-binary.md]:
```suggestion
Then, execute the following command in the terminal to run DM-worker:
```

File: dev/how-to/deploy/data-migration-with-binary.md [hunk 10]
```diff
@@ -0,0 +1,367 @@
...
+
+{{< copyable "shell-regular" >}}
+
+```bash
+bin/dm-worker -config conf/dm-worker1.toml
+```
+
+In DM-worker2, change `source-id` in the configuration file to `mysql-replica-02` and change the `from` configuration part to the address of MySQL2.
```
Comment (yikeke) [inline dev/how-to/deploy/data-migration-with-binary.md]:
```suggestion
In DM-worker2, change `source-id` in the configuration file to `mysql-replica-02` and change the `from` configuration to the address of MySQL2.
```

File: dev/how-to/deploy/data-migration-with-binary.md [hunk 11]
```diff
@@ -0,0 +1,367 @@
...
+
+### Deploy DM-master
+
+You can configure DM-worker by using command-line parameters or the configuration file.
```
Comment (yikeke) [inline dev/how-to/deploy/data-migration-with-binary.md]:
```suggestion
You can configure DM-master by using command-line parameters or the configuration file.
```

File: dev/how-to/deploy/data-migration-with-binary.md [hunk 12]
```diff
@@ -0,0 +1,367 @@
...
+
+```bash
+./bin/dm-master --help
+```
+
+```
+Usage of dm-master:
+  -L string
+        Log level. Available values: "debug", "info" (default value), "warn", "error" or "fatal"
+  -V    The output version
```
Comment (yikeke) [inline dev/how-to/deploy/data-migration-with-binary.md]:
```suggestion
  -V    Outputs the version information
```

File: dev/how-to/deploy/data-migration-with-binary.md [hunk 13]
```diff
@@ -0,0 +1,367 @@
...
+  -config string
+        The path of the configuration file
+  -log-file string
+        The path of log files
+  -master-addr string
+        DM-master address
+  -print-sample-config
+        Prints the example configuration of DM-master
```
Comment (yikeke) [inline dev/how-to/deploy/data-migration-with-binary.md]:
```suggestion
        Prints the sample configuration of DM-master
```
To unify the wording

File: dev/how-to/deploy/data-migration-with-binary.md [hunk 14]
```diff
@@ -0,0 +1,367 @@
...
+> In some situations, you cannot use the above method to configure DM-worker because some configurations are not exposed to the command line. Then use the configuration file instead.
+
+**Deployment method 2: configuration file**
+
+Below is the DM-worker configuration file. It is recommended that you use this method and write the following configuration to `conf/dm-worker1.toml`.
+
+```toml
+# Worker Configuration.
+
+# log configuration
```
Comment (yikeke) [inline dev/how-to/deploy/data-migration-with-binary.md]:
```suggestion
# Log configurations
```

File: dev/how-to/deploy/data-migration-with-binary.md [hunk 15]
```diff
@@ -0,0 +1,367 @@
...
+rpc-rate-burst = 40
+
+# Log configurations
+log-level = "info"
+log-file = "dm-master.log"
+
+# The listening address of DM-master
+master-addr = ":8261"
+
+# replication group <-> DM-worker configuration
```
Comment (yikeke) [inline dev/how-to/deploy/data-migration-with-binary.md]:
```suggestion
# Replication group <-> DM-worker configuration
```
This `<->` symbol means "refers to DM-worker configuration section"?

File: dev/how-to/deploy/data-migration-with-binary.md [hunk 16]
```diff
@@ -0,0 +1,367 @@
...
+dm-worker = "192.168.0.5:8262"
+
+[[deploy]]
+# Corresponding to the source-id in the DM-worker2 configuration file
+source-id = "mysql-replica-02"
+# The service address of DM-worker1
+dm-worker = "192.168.0.6:8262"
+```
+
+Then, run the following command in the terminal to operate DM-master:
```
Comment (yikeke) [inline dev/how-to/deploy/data-migration-with-binary.md]:
```suggestion
Then, execute the following command in the terminal to run DM-master:
```

File: dev/how-to/deploy/data-migration-with-binary.md [hunk 17]
```diff
@@ -0,0 +1,367 @@
...
+
+```bash
+bin/dm-master -config conf/dm-master.toml
+```
+
+Now, a DM cluster is successfully deployed.
+
+### Create a data replication task
+
+Suppose that MySQL1 and MySQL2 have several sharded tales on each instance. These tables have the same structure and the same prefix "t" in their table names. The databases that they are in are named with the same prefix "sharding". In each sharded table, the primary key and unique key are different from those of all other tables.
```
Comment (yikeke) [inline dev/how-to/deploy/data-migration-with-binary.md]:
```suggestion
Suppose that there are several sharded tales on both MySQL1 and MySQL2 instances. These tables have the same structure and the same prefix "t" in their names. The databases where they are located are named with the same prefix "sharding". In each sharded table, the primary key and unique key are different from those of all other tables.
```

File: dev/how-to/deploy/data-migration-with-binary.md [hunk 18]
```diff
@@ -0,0 +1,367 @@
+---
+title: Deploy Data Migration Cluster Using DM Binary
+summary: Learn how to deploy the Data Migration cluster using DM binary.
+category: how-to
+---
+
+# Deploy Data Migration Cluster Using DM Binary
+
+<!-- markdownlint-disable MD029 -->
```
Comment (yikeke) [inline dev/how-to/deploy/data-migration-with-binary.md]:
```suggestion
```
Why disabled this rule for this doc?

File: dev/how-to/deploy/data-migration-with-binary.md (hunk 15 above)
Comment (TomShawn) [reply dev/how-to/deploy/data-migration-with-binary.md]: I have confirmed with @WangXiangUSTC  that it is a typo.

File: dev/how-to/deploy/data-migration-with-binary.md [hunk 19]
```diff
@@ -244,124 +242,124 @@ Now, a DM cluster is successfully deployed.
...
+    sharding-route-rules-table:
+        schema-pattern: sharding*
+        table-pattern: t*
+        target-schema: db_target
+        target-table: t_target
+
+    sharding-route-rules-schema:
+        schema-pattern: sharding*
+        target-schema: db_target
+
```
Comment (yikeke) [inline dev/how-to/deploy/data-migration-with-binary.md]: The relative indentation in this YAML file is wrong. Please check the original indentation and fix it.

**Comment by TomShawn** [comment]
@yikeke  PTAL again, thanks.

**Pull request merged**